packages/dataviews/src/view-table.js

```diff
@@ -366,14 +401,13 @@ function ViewTable( {
 								) }
 							</tr>
 						) ) }
-					</tbody>
-				</table>
-			) }
-			{ ! hasData && (
-				<div className="dataviews-no-results">
-					<p>{ __( 'No results' ) }</p>
-				</div>
-			) }
+				</tbody>
+			</table>
+			<div className="dataviews-no-results" id={ tableNoticeId }>
```

Comment:
Given this component is no longer about "no results" but holds any message related to the table (loading, no results), the class could be renamed to `dataviews-table-notice` or something along those lines.